File: python/7_kyu/count_the_digit.py
# Count the Digit
# https://www.codewars.com/kata/566fc12495810954b1000030/train/python

def nb_dig(n, d):
    lis = []
    for i in range(0, n+1):
        lis.append(str(i**2))
    
    string_lis = ''.join(lis)
    string_lis = list(string_lis)
    
    count = 0
    for i in string_lis:
        if i == str(d):
            count += 1
    
    return count

File: javascript/7_kyu/count_the_digit.js
// Count the Digit
// https://www.codewars.com/kata/566fc12495810954b1000030/train/javascript

function nbDig(n, d) {
    var ans = [];

    for (var i = 0; i <= n; i++) {
        var num = i**2;
        ans.push(num.toString());
    }
    
    var res = ans.join('');
    var count = 0;

    for (var j = 0; j < res.length; j++) {
        if (res[j] == d.toString()) {
            count++;
        }
    }
    return count;
}

console.log(nbDig(10, 1));